app/models/utils/nav/customization_parser.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Utils
+  module Nav
+    # Parses the top menu customization setting
+    class CustomizationParser
+      attr_accessor :markdown_string
+
+      def initialize(markdown_string)
+        self.markdown_string = markdown_string
+      end
+
+      def parse
+        settings_topmenu = markdown_string || ""
```

Comment:
This kind of thing can go in the constructor.

Reply:
I'm not sure what you mean here.  Are you saying to do the parsing in the constructor, and return the result?

Reply:
Typically adding a default value if none is given is done in the constructor, while the actual logic of parsing happens in the parse method. But either is fine.